sdks/go/cmd/specialize/main.go

```diff
@@ -238,3 +246,133 @@ func upto(i int) []int {
 	}
 	return ret
 }
+
+func add(i int, j int) int {
+	return i + j
+}
+
+func mult(i int, j int) int {
+	return i * j
+}
+
+func dict(values ...interface{}) map[string]interface{} {
+	dict := make(map[string]interface{}, len(values)/2)
+	if len(values)%2 != 0 {
+		panic("Invalid dictionary call")
+	}
+	for i := 0; i < len(values); i += 2 {
+		dict[values[i].(string)] = values[i+1]
+	}
+
+	return dict
+}
+
+func list(values ...string) []string {
+	return values
+}
+
+func genericTypingRepresentation(in int, out int, includeType bool) string {
+	seenElements := false
+	typing := ""
+	if in > 0 {
+		typing += fmt.Sprintf("[I%v", 0)
+		for i := 1; i < in; i++ {
+			typing += fmt.Sprintf(", I%v", i)
+		}
+		seenElements = true
+	}
+	if out > 0 {
+		i := 0
+		if !seenElements {
+			typing += fmt.Sprintf("[R%v", 0)
+			i++
+		}
+		for i < out {
+			typing += fmt.Sprintf(", R%v", i)
+			i++
+		}
+		seenElements = true
+	}
+
+	if seenElements {
+		if includeType {
+			typing += " any"
+		}
+		typing += "]"
+	}
+
+	return typing
+}
+
+func possibleBundleLifecycleParameterCombos(numInInterface interface{}, processElementInInterface interface{}) [][]string {
+	numIn := numInInterface.(int)
+	processElementIn := processElementInInterface.(int)
+	ordered_known_parameter_options := []string{"context.Context", "typex.PaneInfo", "[]typex.Window", "typex.EventTime", "typex.BundleFinalization"}
+	// Because of how Bundle lifecycle functions are invoked, all known parameters must preced unknown options and be in order.
+	// Once we hit an unknown options, all remaining unknown options must be included since all iters/emitters must be included
+	// Therefore, we can generate a powerset of the known options and fill out any remaining parameters with an ordered set of remaining unknown options
+	pSetSize := int(math.Pow(2, float64(len(ordered_known_parameter_options))))
+	combos := make([][]string, 0, pSetSize)
+
+	var index int
+	for index < pSetSize {
```

Comment:
There doesn't seem to be any reason not to use a standard `for index := 0; index < pSetSize; index++ {` construct here.

(Don't forget to remove the `index++` from the bottom of the loop.)

Reply:
There's not - this was a consequence of an evolving approach and I missed this bit of cleanup - thanks